### ISSUE TITLE
False `UndefinedMagicMethod`: `Schema\IndexDefinition` fluent modifiers (language(), etc.) not resolved

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1217]
## Description

`Illuminate\Database\Schema\IndexDefinition` fluent modifiers (e.g. `language()`) are reported as false `UndefinedMagicMethod`. Laravel declares these as class-level `@method` annotations on `IndexDefinition` (which extends `Fluent`), so `$table->fullText(...)->language(...)` is valid and works at runtime.

## Reproduction

Found auditing [koel/koel](https://github.com/koel/koel), `database/migrations/..._tune_fulltext_search_indexes.php`:

```php
Schema::table('songs', function (Blueprint $table) {
    $table->fullText(['title', 'artist_name', 'album_name'])->language('simple');
});
```

## Actual behavior

```
UndefinedMagicMethod: Magic method Illuminate\Database\Schema\IndexDefinition::language does not exist
```

(7 occurrences across the migration.)

## Expected behavior

No error — `language()` is a documented fluent modifier on `IndexDefinition`.

## Cause

Laravel source (`Illuminate/Database/Schema/IndexDefinition.php`):

```php
/**
 * @method $this algorithm(string $algorithm) ...
 * @method $this language(string $language) Specify a language for the full text index (PostgreSQL)
 * @method $this deferrable(bool $value = true) ...
 * @method $this initiallyImmediate(bool $value = true) ...
 */
class IndexDefinition extends Fluent { }
```

`Blueprint::fullText()`/`index()`/`spatialIndex()` return `IndexDefinition`, whose fluent methods live only in the class `@method` block. The plugin should either (a) let Psalm read these `@method` tags from the reflected class, or (b) if it stubs `IndexDefinition`, carry the full `@method` list verbatim (a bare `class IndexDefinition { }` stub would wipe them — cf. the "class declarations wipe reflected metadata" note in CONTRIBUTING).

## Proposed fix

Add/verify a stub for `IndexDefinition` that preserves the `@method` fluent modifiers (`algorithm`, `language`, `deferrable`, `initiallyImmediate`), and confirm `Blueprint::fullText()`/`index()` return types point at it.
